### PR TITLE
Added worldgen for stones

### PR DESCRIPTION
**Pull request by ConnorRowe**
I'm sure there is a better method for finding the blocks to generate but I couldn't figure one out sorry. If anyone has a problem with anything tell me and I'll fix it. Thanks.

### CODE REVIEW

File: src/main/java/team/chisel/common/util/GenerationHandler.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            
+            int basaltHeight = 35;
+            int marbleHeight = 100;
+            int limestoneHeight = 40;
+            
+            //Basalt Generation
+            int basaltY = rand.nextInt(basaltHeight);
+            BlockPos basaltPos = new BlockPos(firstBlockXCoord, basaltY, firstBlockZCoord);            
+            if (Configurations.basaltAmount > 0)
+            	(new WorldGenMinable(ForgeRegistries.BLOCKS.getValue(new ResourceLocation("chisel", "basaltextra")).getStateFromMeta(7), Configurations.basaltAmount)).generate(world, rand, basaltPos);
```
Comment (tterrag1098) [inline src/main/java/team/chisel/common/util/GenerationHandler.java]: Don't do this lookup every time, cache it in a field.

Also, add a //TODO for when we do finally have an easy Block reference class

File: src/main/java/team/chisel/common/util/GenerationHandler.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraft.world.chunk.IChunkGenerator;
+import net.minecraft.world.chunk.IChunkProvider;
+import net.minecraft.world.gen.feature.WorldGenMinable;
+import net.minecraftforge.fml.common.IWorldGenerator;
+import net.minecraftforge.fml.common.registry.ForgeRegistries;
+import team.chisel.common.config.Configurations;
+
+public class GenerationHandler implements IWorldGenerator {
```
Comment (tterrag1098) [inline src/main/java/team/chisel/common/util/GenerationHandler.java:14]: Would be worthwhile to extend WorldGenMineable which has code for only replacing stone.

File: src/main/java/team/chisel/common/util/GenerationHandler.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	public void generate(Random random, int chunkX, int chunkZ, World world, IChunkGenerator chunkGenerator, IChunkProvider chunkProvider) {
+		switch (world.provider.getDimension()) {
+		case -1:
+			break;
+		case 0:
+			generateSurface(world, random, chunkX * 16, chunkZ * 16);
+			break;
+		case 1:
+			break;
+		}
```
Comment (Drullkus) [inline src/main/java/team/chisel/common/util/GenerationHandler.java:28]: I'd like to be able to configure generation in other worlds, too...

Like Basalt in the nether.

**Comment by ConnorRowe** [comment]
Okay working on it :)

**Comment by tterrag1098** [comment]
Also, there is a block reference class here https://github.com/Chisel-Team/Chisel/blob/1.9/dev/src/main/java/team/chisel/common/init/ChiselBlocks.java

Add any you need if they are missing.

**Comment by ConnorRowe** [comment]
I'm not quite sure how to generate the different stones when it extends WorldGenMinable, anyone have any tips?

Edit: Also, does anyone have any preferences on what Y-levels the stones should generate on, and how common they should be?

**Comment by jriwanek** [comment]
Take a look at:
https://github.com/cyanobacterium/OreSpawn

**Comment by tterrag1098** [comment]
Merged manually via 2b0411de0ba4c42ea29bd976dbd466c017909eaf

**Pull request closed without merging**